thunder/tests/test_jit_general.py

```diff
@@ -1090,3 +1090,32 @@ def forward(self, x):
     actual = thunder.jit(m)(x)
 
     torch.testing.assert_close(actual, expected)
+
+
+@pytest.mark.parametrize(
+    "device",
+    ("cpu", "cuda"),
+)
+def test_cache_symbolic_values_reshape(device):
+    if not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    a = torch.randn((4, 8, 6), device=device)
+
+    def foo(t, batch_size):
+        return t.reshape(batch_size, -1).sum(-1)
```

Comment:
we might want to discuss later on how we would like to represent dynamic shape. (design review due)

Meanwhile, this is a quick patch coming from patching @kiya00 's issue: https://github.com/Lightning-AI/lightning-thunder/pull/451#issuecomment-2189428679

Reply:
Thanks for the fix, I'll try it once it merged